### ISSUE TITLE
Logits issue with transformers >4.51.0

### ISSUE THREAD

**Issue by jaideepr97** [instructlab/instructlab#3289]
**Describe the bug**
There is a bug within the newer transformers releases (>4.51.0) that results in a consistent failure within `linux_train.py`
The training process fails somewhere inside of the transformers library call to generate samples and provides no stack trace, other than saying that a tuple object is generated somewhere that does not contain any logits

Updating parameters to the function call does not seem to fix the problem 
This issue only affects linux_train.py which is legacy code at this point

**To Reproduce**
1. Try to execute simple train pipeline in a linux environment or run small-test job in CI 
2. hit error

**Expected behavior**

**Screenshots**
```
LINUX_TRAIN.PY: SANITY CHECKING THE BASE MODEL
after tokenizer call
before generate call
'tuple' object has no attribute 'logits'
  0%|          | 0/5 [00:00<?, ?it/s]
  0%|          | 0/5 [00:00<?, ?it/s]
+ rm -rf /home/tmp/tmp.O2La1qkAMT
Error: Process completed with exit code 1.
```

**Device Info (please complete the following information):**
 - Hardware Specs: [e.g. Apple M2 Pro Chip, 16 GB Memory, etc.]
 - OS Version: [e.g. Mac OS 14.4.1, Fedora Linux 40]
 - Python Version: [output of `python --version`]
 - InstructLab Version: [output of `ilab system info`]

**Additional context**

**Comment by jpodivin** [comment]
Instructlab version 0.26 is affected by this issue.

**Comment by jpodivin** [comment]
I did some research on this. Exception is raised from [transformers utils.py](https://github.com/huggingface/transformers/blob/main/src/transformers/generation/utils.py#L3434 ), it's only obfuscated by lack of trace.

While I'm not certain what exactly is going wrong, I did find a workaround.

```
diff --git a/src/instructlab/train/linux_train.py b/src/instructlab/train/linux_train.py
index b651251f..0e0f218b 100644
--- a/src/instructlab/train/linux_train.py
+++ b/src/instructlab/train/linux_train.py
@@ -212,7 +212,7 @@ def linux_train(
     # Loading the model
     print("LINUX_TRAIN.PY: LOADING THE BASE MODEL")
     config = AutoConfig.from_pretrained(
-        model_name, torchscript=True, trust_remote_code=True
+        model_name, trust_remote_code=True
     )
 
     # https://huggingface.co/docs/transformers/en/model_doc/auto#transformers.AutoModelForCausalLM.from_pretrained
```

When I remove `torchscript` parameter from config, I can execute following step of the process. [Torchscript](https://pytorch.org/tutorials/beginner/Intro_to_TorchScript_tutorial.html) is meant to speed python up with some JIT magic. 

Unfortunately, it seems that transformers use of this technique left some edge cases unresolved.

**Comment by booxter** [comment]
This is fixed by https://github.com/instructlab/instructlab/pull/3307 ; if there are performance or other concerns with removing torchscript, these should be dealt in a new bug report.
